### PR TITLE
Patch modified gates with QASM2 implementations.

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Add an internal version of `stdgates.inc` with patched gate definitions.

### CODE REVIEW

**Review by lcauser-oqc — APPROVED** [review]
LGTM, thanks! :smile:

**Review by daria-oqc — APPROVED** [review]

File: src/qat/purr/integrations/qasm.py [hunk 1]
```diff
@@ -988,10 +990,13 @@ def visit_QuantumGate(self, node: ast.QuantumGate, context: QasmContext):
             case "ecr" | "ECR":
                 return self.add_ecr(target_qubits, self.builder)
 
+        if len(node.modifiers) > 0:
```
Comment (daria-oqc) [inline src/qat/purr/integrations/qasm.py:993]: Thanks!

File: src/qat/purr/integrations/grammars/stdgates.inc [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+gate z a { p(π) a; }
+
+// Clifford gate: Hadamard
+gate h a { U(π/2, 0, π) a; gphase(-π/4);}
+// Clifford gate: sqrt(Z) or S gate
+// gate s a { pow(0.5) @ z a; }
+gate s a { u1(pi/2) a; }
+// Clifford gate: inverse of sqrt(Z)
+// gate sdg a { inv @ pow(0.5) @ z a; }
+gate sdg a { u1(-pi/2) a; }
```
Comment (rrundle-oqc) [inline src/qat/purr/integrations/grammars/stdgates.inc:29]: likewise, this needs a  gphase(+np.pi/4)

File: src/qat/purr/integrations/grammars/stdgates.inc [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+// sqrt(S) or T gate
+// gate t a { pow(0.5) @ s a; }
+gate t a { u1(pi/4) a; }
```
Comment (rrundle-oqc) [inline src/qat/purr/integrations/grammars/stdgates.inc:33]: gphase(-np.pi/8)

File: src/qat/purr/integrations/grammars/stdgates.inc [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+// inverse of sqrt(S)
+// gate tdg a { inv @ pow(0.5) @ s a; }
+gate tdg a { u1(-pi/4) a; }
```
Comment (rrundle-oqc) [inline src/qat/purr/integrations/grammars/stdgates.inc:36]: gphase(np.pi/8)

File: src/qat/purr/integrations/grammars/stdgates.inc [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+// sqrt(NOT) gate
+// gate sx a { pow(0.5) @ x a; }
+gate sx a { sdg a; h a; sdg a; }
```
Comment (elskysr) [inline src/qat/purr/integrations/grammars/stdgates.inc:41]: where is this re-defintion taken from?

**Review by lcauser-oqc — APPROVED** [review]

File: src/qat/purr/integrations/grammars/stdgates.inc (hunk 5 above)
Comment (keriksson-rosenqvist) [reply src/qat/purr/integrations/grammars/stdgates.inc:41]: All the re-definitions are taken from the QASM2 version: https://github.com/Qiskit/qiskit/blob/main/qiskit/qasm/libs/qelib1.inc

**Pull request merged**